packages/cells/src/widget.ts

```diff
@@ -1325,15 +1327,28 @@ export abstract class AttachmentsCell extends Cell {
       const mimeType = matches[1];
       const encodedData = matches[3];
       const bundle: nbformat.IMimeBundle = { [mimeType]: encodedData };
-      this.model.attachments.set(blob.name, bundle);
-      this.updateCellSourceWithAttachment(blob.name);
+      const URI = this._generateURI(blob.name);
+
+      if (mimeType.startsWith('image/')){
+        this.model.attachments.set(URI, bundle);
+        this.updateCellSourceWithAttachment(name, URI);
```

Comment:
I think this should be `blob.name` instead of the global `name`.

```suggestion
        this.updateCellSourceWithAttachment(blob.name, URI);
```